src/array/cuda/array_index_select.cu

```diff
@@ -33,9 +27,13 @@ NDArray IndexSelect(NDArray array, IdArray index) {
 
   // use index->ctx for pinned array
   NDArray ret = NDArray::Empty(shape, array->dtype, index->ctx);
-  if (len == 0) return ret;
+  if (len == 0 || arr_len == 0) return ret;
```

Comment:
Just wondering if this checking condition is complete or not... It seems only checking the length of the first dim, right? Could it cover the case when `array->shape[i] == 0`?

Reply:
I cannot imagine a tensor has other dims when the first dim is 0.

Reply:
Here is the utility to check a null array.
https://github.com/dmlc/dgl/blob/e28f07812bae274f5ba49d26629de5541c9cff60/include/dgl/aten/array_ops.h#L37

Reply:
I see...

Reply:
I get your point. It would be better to use `array.NumElements()` here.

Reply:
Or simply `arr_len * num_feat` here.